core/finality-grandpa/src/communication/gossip.rs

```diff
@@ -1183,6 +1255,13 @@ impl<Block: BlockT> network_gossip::Validator<Block> for GossipValidator<Block>
 				Some(x) => x,
 			};
 
+            if let MessageIntent::Broadcast { previous_attempts } = intent {
+                // early return if the message isn't allowed at this stage.
+                if !inner.message_allowed(peer, previous_attempts) {
```

Comment:
This filter is currently applied before the GRANDPA politeness filter, should we do it afterwards instead?

Reply:
This check is quicker, isn't it?